src/cascadia/TerminalSettingsModel/WslDistroGenerator.cpp

```diff
@@ -136,3 +136,206 @@ std::vector<Profile> WslDistroGenerator::GenerateProfiles()
+    DWORD dwNumSubKeys = 0;
+    DWORD maxSubKeyLen = 0;
+    if (RegQueryInfoKey(wslRootKey.get(), // hKey,
+                        nullptr, // lpClass,
+                        nullptr, // lpcchClass,
+                        nullptr, // lpReserved,
+                        &dwNumSubKeys, // lpcSubKeys,
+                        &maxSubKeyLen, // lpcbMaxSubKeyLen,
+                        nullptr, // lpcbMaxClassLen,
+                        nullptr, // lpcValues,
+                        nullptr, // lpcbMaxValueNameLen,
+                        nullptr, // lpcbMaxValueLen,
+                        nullptr, // lpcbSecurityDescriptor,
+                        nullptr // lpftLastWriteTime
+                        ) != ERROR_SUCCESS)
+    {
+        return false;
+    }
+
+    // lpcbMaxSubKeyLen does not include trailing space
+    std::unique_ptr<wchar_t[]> buffer = std::make_unique<wchar_t[]>(maxSubKeyLen + 1);
+
+    // reserve enough space for all the GUIDs we're about to enumerate
+    guidStrings.reserve(dwNumSubKeys);
+    for (DWORD i = 0; i < dwNumSubKeys; i++)
+    {
+        auto cbName = maxSubKeyLen + 1;
+        const auto result = RegEnumKeyEx(wslRootKey.get(), i, buffer.get(), &cbName, nullptr, nullptr, nullptr, nullptr);
+        if (result == ERROR_SUCCESS)
+        {
+            guidStrings.emplace_back(buffer.get(), cbName);
+        }
+    }
+    return true;
+}
+
+// Function Description:
+// - Get the list of all the names of all the WSL distros from the registry. If
+//   we fail to open any regkey for the GUID of a distro, we'll just skip it.
+//   Places the names of all the distros into the "names" param.
+// Arguments:
+// - wslRootKey: the HKEY for the Lxss node.
+// - guidStrings: A list of all the GUIDs of the installed distros
+// - names: a vector that recieves all the names of the installed distros.
+// Return Value:
+// - false if the root key was invalid, else true.
+bool _getWslNames(const wil::unique_hkey& wslRootKey,
+                  std::vector<std::wstring>& guidStrings,
+                  std::vector<std::wstring>& names)
+{
+    if (!wslRootKey)
+    {
+        return false;
+    }
+    for (const auto& guid : guidStrings)
+    {
+        wil::unique_hkey distroKey{ _openDistroKey(wslRootKey, guid) };
+        if (!distroKey)
+        {
+            // return false;
+            continue;
+        }
+
+        DWORD bufferCapacity = 0;
+        auto result = RegQueryValueEx(distroKey.get(), RegKeyDistroName, 0, nullptr, nullptr, &bufferCapacity);
+        // request regkey binary buffer capacity only
+        if (result != ERROR_SUCCESS)
+        {
+            // return false;
+            continue;
+        }
+        std::unique_ptr<wchar_t[]> buffer = std::make_unique<wchar_t[]>(bufferCapacity);
+        // request regkey binary content
+        result = RegQueryValueEx(distroKey.get(), RegKeyDistroName, 0, nullptr, reinterpret_cast<BYTE*>(buffer.get()), &bufferCapacity);

```

Comment:
You can write this and drop the extra size-check above:
```cpp
std::wstring buffer;
result = wil::AdaptFixedSizeToAllocatedResult<std::wstring, 256>(buffer, [&](PWSTR value, size_t valueLength, size_t* valueLengthNeededWithNull) -> HRESULT {
	auto length = static_cast<DWORD>(valueLength);
    const auto hr = RegQueryValueExW(distroKey.get(), RegKeyDistroName, 0, nullptr, value, &length);
	*valueLengthNeededWithNull = length + 1;
	return hr == ERROR_MORE_DATA ? S_OK : hr;
});
names.emplace_back(std::move(buffer));
```

Reply:
LOVE this thing